jdaviz/configs/imviz/plugins/catalogs/catalogs.py

```diff
@@ -0,0 +1,121 @@
+            self.catalog_selected = self.catalog_items[0]
+
+    def search(self):
+        # note to self: should I call clear before searching?
+
+        # gets the current viewer
+        viewer = self.viewer.selected_obj
+
+        # nothing happens in the case there is no image in the viewer
+        if viewer.state.reference_data is None:
+            self.results_available = False
+            self.number_of_results = 0
+            return
+        # obtains the center point of the current image and converts the point into sky coordinates
+        x_center = (viewer.state.x_min + viewer.state.x_max) * 0.5
+        y_center = (viewer.state.y_min + viewer.state.y_max) * 0.5
+        skycoord_center = viewer.state.reference_data.coords.pixel_to_world(x_center, y_center)
+
+        # obtains the viewer's zoom limits based on the visible layer
+        ny, nx = viewer.state.reference_data.shape
+        zoom_x_min = max(0, viewer.state.x_min)
+        zoom_x_max = min(nx, viewer.state.x_max)
+        zoom_y_min = max(0, viewer.state.y_min)
+        zoom_y_max = min(ny, viewer.state.y_max)
+        zoom_coordinate = viewer.state.reference_data.coords.pixel_to_world(
+            [zoom_x_min, zoom_x_min, zoom_x_max, zoom_x_max],
+            [zoom_y_min, zoom_y_max, zoom_y_max, zoom_y_min])
+
+        # radius for querying the region is based on:
+        # the distance between the longest zoom limits and the center point
+        zoom_radius = max(skycoord_center.separation(zoom_coordinate))
+
+        # conducts search based on SDSS
+        if self.catalog_selected == "SDSS":
+            # queries the region (based on the provided center point and radius)
+            # finds all the sources in that region
+            query_region_result = SDSS.query_region(skycoord_center, radius=zoom_radius,
+                                                    data_release=17)
+            self.results_available = True
+
+        else:
+            self.results_available = False
+            self.number_of_results = 0
+            raise ValueError(f"{self.catalog_selected} not a supported catalog")
+
+        # nothing happens in the case the query returned empty
+        if query_region_result is None:
+            self.number_of_results = 0
+            return
+
+        # a table is created storing the 'ra' and 'dec' plottable points of each source found
+        skycoord_table = SkyCoord(query_region_result['ra'], query_region_result['dec'], unit='deg')
+
+        # coordinates found are converted to pixel coordinates
+        pixel_table = viewer.state.reference_data.coords.world_to_pixel(skycoord_table)
+        # coordinates are filtered out (using a mask) if outside the zoom range
+        pair_pixel_table = np.dstack((pixel_table[0], pixel_table[1]))
+        # ma.masked_outside removes the coordinates outside the zoom range
+        # ma.compress_rows removes any row that has a mask mark
+        filtered_table = ma.compress_rows(
```

Comment:
Does the user want the original query result back out via API? If so, we can apply the mask from this filter to `query_region_result` too and then have it be API accessible like the aperture photometry results.

Marking objects in viewer is not useful if you don't have all the relevant data that go with those objects (e.g., are they stars or galaxies, what are the magnitudes, etc).

Reply:
I was able to get the table out, but it isn't filtered. The issue with this filtering method above is that it essentially lost all of the important details to make a table of coordinate points for the `add_markers` function to use. I can try seeing if I can apply a mask to `query_region_result` tomorrow without losing all of the extra 'stuff' the querying function table has.

Reply:
Okay, if you haven't already, maybe add a note about that in the code and user doc, so we can follow up in the future. Probably not a big deal as people can always filter the table again themselves.

Reply:
I tried filtering the table, yet somehow with the way I did it in the API, I get even less coordinates compared to what's shown in 'results found'. I checked the points though and those are correct.

Reply:
Update: I wasn't able to filter the table out, and I made note of that in `plugins.rst